### PR TITLE
Fix for libprotobuf 4.25.1

### PR DESCRIPTION
**Pull request by xylar**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

File: recipe/conda_build_config.yaml [hunk 1]
```diff
@@ -1,3 +1,3 @@
 # manual pins because we're linking migrations
 libprotobuf:
-  - 4.24.4
+  - 4.25.1
```
Comment (xylar) [inline recipe/conda_build_config.yaml:3]: Is it better to keep this an bump it or remove it?

**Comment by xylar** [comment]
@conda-forge-admin, please rerender

File: recipe/conda_build_config.yaml (hunk 1 above)
Comment (h-vetinari) [reply recipe/conda_build_config.yaml:3]: No, I should have updated it... 🤦

File: recipe/meta.yaml [hunk 2]
```diff
@@ -34,7 +34,7 @@ source:
     - patches/0008-link-grpc-_unsecure-to-grpc.patch
 
 build:
-  number: 0
+  number: 1
```
Comment (h-vetinari) [inline recipe/meta.yaml:37]: Can you back out the build number bump please? It seems the migrator took precedence over the local cbc anyway, so updating the CBC doesn't actually change anything about the builds (though it's good to keep it up-to-date)

Comment (xylar) [reply recipe/meta.yaml]: I don't fully follow but I'll defer to your expertise here.

**Comment by xylar** [comment]
@h-vetinari, could you have another look when you have time?

**Review by h-vetinari — APPROVED** [review]

**Pull request merged**